components/org.wso2.carbon.identity.outbound.provisioning.connector.office365/src/main/java/org/wso2/carbon/identity/outbound/provisioning/connector/office365/Office365ProvisioningConnector.java

```diff
@@ -0,0 +1,420 @@
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpDelete;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.client.methods.HttpRequestBase;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.apache.http.message.BasicNameValuePair;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.json.JSONTokener;
+import org.wso2.carbon.identity.application.common.model.Property;
+import org.wso2.carbon.identity.base.IdentityConstants;
+import org.wso2.carbon.identity.core.util.IdentityUtil;
+import org.wso2.carbon.identity.provisioning.AbstractOutboundProvisioningConnector;
+import org.wso2.carbon.identity.provisioning.IdentityProvisioningConstants;
+import org.wso2.carbon.identity.provisioning.IdentityProvisioningException;
+import org.wso2.carbon.identity.provisioning.ProvisionedIdentifier;
+import org.wso2.carbon.identity.provisioning.ProvisioningEntity;
+import org.wso2.carbon.identity.provisioning.ProvisioningEntityType;
+import org.wso2.carbon.identity.provisioning.ProvisioningOperation;
+
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+
+/**
+ * This class handles the Office365 user provisioning operations to Azure AD.
+ */
+public class Office365ProvisioningConnector extends AbstractOutboundProvisioningConnector {
+
+    private static final Log log = LogFactory.getLog(Office365ProvisioningConnector.class);
+    private Office365ProvisioningConnectorConfig configHolder;
+
+    @Override
+    public void init(Property[] provisioningProperties) throws IdentityProvisioningException {
+
+        Properties configs = new Properties();
+
+        if (provisioningProperties != null && provisioningProperties.length > 0) {
+            for (Property property : provisioningProperties) {
+                configs.put(property.getName(), property.getValue());
+                if (IdentityProvisioningConstants.JIT_PROVISIONING_ENABLED.equals(property
+                        .getName())) {
+                    if (Office365ConnectorConstants.PROPERTY_VALUE_TRUE.equals(property.getValue())) {
+                        jitProvisioningEnabled = true;
+                    }
+                }
+            }
+        }
+
+        configHolder = new Office365ProvisioningConnectorConfig(configs);
+    }
+
+    @Override
+    public ProvisionedIdentifier provision(ProvisioningEntity provisioningEntity)
+            throws IdentityProvisioningException {
+
+        String provisionedId = null;
+
+        if (provisioningEntity != null) {
+
+            if (provisioningEntity.isJitProvisioning() && !isJitProvisioningEnabled()) {
+                log.debug("JIT provisioning disabled for Office365 connector");
```

Comment:
Shall we put debug logs inside an if condition which checks if debug is enabled? (ex: log.isdebugEnabled())

Reply:
Fixed with https://github.com/wso2-extensions/identity-office365/pull/5